### DISCUSSION TITLE
Use pkgxdev for private repository

### DISCUSSION THREAD

**Discussion by rinshadka** [pkgxdev/pkgx#852] [Help]
Hi,

Can I use pkgxdev for my own private repository build process and push the package to pkgx pantry , OR pkgdev pantry only works with opensource projects and a verification and merge process is there?

Also, Can I create my own pkgx pantry self hosted for keeping the packages of my private projects.

Please let me know.

Thanks.

**Comment by mxcl** [comment]
hi, we had a discussion similar to this someplace but I can't find it.

I should think this is possible. `PKGX_PANTRY_PATH` is a variable you can set that adds additional pantries to the path. They can contain anything you like. You can then have the `package.yml` for your own packages install your own binaries.

Main issue is that currently the place we grab bottles (the built products) is hardcoded to `dist.pkgx.dev`. Probs we can figure out a quick solution for that where the pantry says where to fetch bottles and how. It could be a bit of code that runs so you can make it dynamic based on the package.

We could add this to a near term minor release if it sounds like the right thing for you…?

Adding to `PKGX_PANTRY_PATH` would remain manual for now. But I think a future where you can `pkgx foo.com/bar` and that will check foo.com for a pantry before installing bar would be neat.

**Reply by rinshadka** [reply]
Thanks @mxcl for the reply. So what I understood now is that we can make use of the `PKGX_PANTRY_PATH` variable inside pkgx.xml file and point this path to a  location in my filesystem and where I wanted the binary output for the respective package.yml file at pantry/projects. Please correct me if I am wrong. Also the authentication to fetch the distributable -> url in package.xml works for a private repo?

Thank You.

**Reply by mxcl** [reply]
currently the bottles (binaries) can only come from https://dist.pkgx.dev so at this time you cannot really have private “taps”.

I would like to provide this feature ASAP but I haven't yet had the inspiration for how it works in a sensible fashion.

**Comment by adam50200** [comment]
What is pkgx?

**Reply by jhheider** [reply]
https://docs.pkgx.sh/#what-is-pkgx
